src/cares_wrap.h

```diff
@@ -304,19 +306,17 @@ class QueryWrap final : public AsyncWrap {
     return wrap;
   }
 
-  static void Callback(
-      void* arg,
-      int status,
-      int timeouts,
-      unsigned char* answer_buf,
-      int answer_len) {
+  static void Callback(void* arg,
+                       ares_status_t status,
+                       size_t timeouts,
+                       const ares_dns_record_t* dnsrec) {
     QueryWrap<Traits>* wrap = FromCallbackPointer(arg);
     if (wrap == nullptr) return;
 
     unsigned char* buf_copy = nullptr;
+    size_t answer_len = 0;
     if (status == ARES_SUCCESS) {
-      buf_copy = node::Malloc<unsigned char>(answer_len);
-      memcpy(buf_copy, answer_buf, answer_len);
+      ares_dns_write(dnsrec, &buf_copy, &answer_len);
```

Comment:
I think this might be causing a memory leak or having some kind of memory problems. The documentation says that the buffer must be cleared using [ares_free_string](https://c-ares.org/docs/ares_free_string.html)..

Reply:
This was the reason for my "rubber stamp", but luckily Matteo verified and read the documentation.

Reply:
It's not needed. Under-the-hood [`ares_free_string` is a wrapper around `free`](https://github.com/c-ares/c-ares/blob/a70637c67ba2657adb54b077f4f8434f00ecbf83/src/lib/ares_library_init.c#L62), which is already called by when [`MallocedBuffer`](https://github.com/nodejs/node/blob/1ba8fbe4b82aea8a4e681a5cff6f47521561970d/src/util.h#L643) deconstructs.

Reply:
Can you please add a comment. It's important to say that if any time we change the allocator in ares, we need to take a look if this is leaking.